### PR TITLE
[Picture] - add picture component

### PR DESCRIPTION
**Pull request by pietro-maximoff**
https://sovryn.monday.com/boards/2218344956/pulses/2236481178

### CODE REVIEW

File: src/app/components/Picture/index.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import classNames from 'classnames';
+import React, { useMemo } from 'react';
+import styles from './index.module.scss';
+import { Breakpoint, BreakpointWidths } from 'types/tailwind';
+
+type Sources = {
+  src: string;
+  media: Breakpoint;
```
Comment (ShynRou) [inline src/app/components/Picture/index.tsx:8]: media is not required.
```suggestion
  media?: Breakpoint;
```

File: src/app/components/Picture/index.module.scss [hunk 2]
```diff
@@ -0,0 +1,5 @@
+.picture {
+    img {
+        max-width: 100%;
```
Comment (ShynRou) [inline src/app/components/Picture/index.module.scss:3]: For this styling the module.scss is not needed. just give the img the className `tw-max-w-full`

File: src/app/components/Picture/index.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+export const Picture: React.FC<PictureProps> = ({
+  src,
+  srcSet,
+  title = 'image',
+  className,
+}) => {
+  const renderSources = useMemo(() => {
+    if (!srcSet) {
+      return <source srcSet={src} />;
```
Comment (ShynRou) [inline src/app/components/Picture/index.tsx:26]: This is redundant, if srcSet is not set we can drop the source elements
```suggestion
      return null;
```

File: src/app/components/Picture/index.tsx [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+type Sources = {
+  src: string;
+  media: Breakpoint;
+};
+
+type PictureProps = {
+  src: string;
+  srcSet?: Sources[];
+  title?: string;
```
Comment (ShynRou) [inline src/app/components/Picture/index.tsx:14]: Sorry, I made a mistake in the spec, title should be called alt here, an not be used as title attribute.
(Have adjusted the spec, now.) I don't think we will need a title property, the alt should be enough.

File: src/app/components/Picture/index.tsx (hunk 3 above)
Comment (pietro-maximoff) [reply src/app/components/Picture/index.tsx]: fixed

File: src/app/components/Picture/index.tsx (hunk 4 above)
Comment (pietro-maximoff) [reply src/app/components/Picture/index.tsx]: np, fixed

File: src/app/components/Picture/index.module.scss (hunk 2 above)
Comment (pietro-maximoff) [reply src/app/components/Picture/index.module.scss]: removed

File: src/app/components/Picture/index.tsx (hunk 1 above)
Comment (pietro-maximoff) [reply src/app/components/Picture/index.tsx]: fixed

File: src/app/components/Picture/index.tsx [hunk 5]
```diff
@@ -0,0 +1,49 @@
+import classNames from 'classnames';
+import React, { useMemo } from 'react';
+import styles from './index.module.scss';
+import { Breakpoint, BreakpointWidths } from 'types/tailwind';
+
+type Sources = {
+  src: string;
```
[the remark this answers is not in the document]
Comment (pietro-maximoff) [reply src/app/components/Picture/index.tsx]: fixed

**Review by ShynRou — CHANGES_REQUESTED** [review]

File: src/app/components/Picture/index.tsx [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+        return <source srcSet={`${sourceArray.join(', ')}`} />;
+      }
+      return (
+        <source
+          key={`source-${index}`}
+          srcSet={`${source.src}`}
+          media={
+            source.media && `(max-width: ${BreakpointWidths[source.media]}px)`
+          }
+        />
```
[the remark this answers is not in the document]
Comment (pietro-maximoff) [reply src/app/components/Picture/index.tsx]: fixed

File: src/app/components/Picture/index.tsx [hunk 7]
```diff
@@ -0,0 +1,66 @@
+import React, { useMemo } from 'react';
+import { Breakpoint, BreakpointWidths } from 'types/tailwind';
+
+type SourceProps = {
+  imageSrc: string;
+  density?: number;
+  width?: Breakpoint;
```
[the remark this answers is not in the document]
Comment (pietro-maximoff) [reply src/app/components/Picture/index.tsx]: done

**Review by tiltom — APPROVED** [review]

**Pull request merged**